Sources/SpeziAccount/AccountSetup.swift

```diff
@@ -253,18 +242,10 @@ public struct AccountSetup<Header: View, Continue: View>: View {
 
     private func handleSetupCompleted(_ details: AccountDetails) {
         setupState = .loadingExistingAccount
-        setupCompleteClosure(details)
-    }
-}
-
-
-extension EnvironmentValues {
-    public var _accountSetupState: _AccountSetupState { // swiftlint:disable:this identifier_name missing_docs
-        get {
-            self[_AccountSetupState.self]
-        }
-        set {
-            self[_AccountSetupState.self] = newValue
+        accountSetupTask?.cancel()
+        accountSetupTask = Task { @MainActor in
+            await setupCompleteClosure(details)
+            setupState = .presentingExistingAccount
```

Comment:
should we check for cancellation before we set the state? Maybe we should also check that this is not called multiple times in parallel in general?

Reply:
You are right. Not allowing it to be called multiple times would probably be the better approach in general. This would then also make checking for cancellation obsolete as, then, it would only happen when the view disappears.